sw/airborne/arch/chibios/modules/uavcan/uavcan.c

```diff
@@ -109,6 +121,26 @@ static void uavcan_tx(void* p)
     pprz_bsem_wait(&iface->bsem);
 
     pprz_mtx_lock(&iface->mutex);
+
+
+    // read the Tx FIFO to canard
+    pprz_mtx_lock(&iface->tx_fifo_mutex);
+    while(true) {
+      struct uavcan_msg_header_t header;
+      int ret = circular_buffer_get(&iface->_tx_fifo, (uint8_t*)&header, sizeof(header));
+      if(ret < 0) {break;}
+      if(header.payload_len >= UAVCAN_MSG_MAX_SIZE) {
+        chSysHalt("UAVCAN_MSG_MAX_SIZE too small");
+      }
+      ret = circular_buffer_get(&iface->_tx_fifo, msg_payload, UAVCAN_MSG_MAX_SIZE);
```

Comment:
Do you always need to get the MAX_SIZE?

Reply:
I changed that a bit, so if the buffer is too small to contains the message, the message is simply discarded. It won't work as expected, but at least it won't crash the autopilot, and it won't affect the other messages.